operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -144,27 +148,23 @@ func (r *TenantReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		retrigErr = err
 	}
 
+	// Determine if the personal namespace should be deleted
+
+	nsName := fmt.Sprintf("tenant-%s", strings.ReplaceAll(tn.Name, ".", "-"))
+
+	// Test if namespace has been open for too long; attempt to delete if there are no instances inside
+	keepNsOpen, err := r.enforceNamespaceKeepAliveOrDelete(ctx, &tn, nsName, r.TenantWorkspaceKeepAlive)
+	if err != nil {
+		klog.Errorf("Error in r.List: unable to capture instances in tenant workspace %s -> %s", nsName, err)
```

Comment:
You should return in case of error, as continuing would lead to an undefined behavior.

Reply:
Resolved in [702078](https://github.com/netgroup-polito/CrownLabs/pull/821/commits/702078c8fde1237b2fe5f7ff478b52983b12bd5a)